source/opt/inline_pass.cpp

```diff
@@ -601,6 +584,48 @@ InlinePass::GetBlocksFunction InlinePass::StructuredSuccessorsFunction() {
   };
 }
 
+bool InlinePass::HasNoReturnInSelection(Function* func) {
```

Comment:
You might be able to use `GetStructuedCFGAnalaysis` in the ir context.  I added it because this same code has already been written too often.  You can get the analysis, and do a single run through of all of the blocks in any order.

Reply:
Done.